src/backend/updates.rs

```diff
@@ -71,19 +78,93 @@ fn post_upgrade() {
 }
 
 async fn post_upgrade_fixtures() {
+    mutate(|state| state.read_only = true);
+    let total_posts = read(|state| state.posts.len());
+
+    // Init all stable structs becasue we do not need the old contents of the stable memory anymore
+    init_stable_structs();
+
+    loop {
+        let num_posts = read(|state| state.posts.len());
+        let _: Result<(), ()> = canisters::call_canister(id(), "migrate_posts", ((),))
+            .await
+            .map_err(|(code, err)| {
+                mutate(|state| {
+                    state.logger.error(format!(
+                        "couldn't migrate posts: code {:?}, err {}",
+                        code, err
+                    ))
+                })
+            });
+        let new_num_posts = read(|state| state.posts.len());
+        if new_num_posts == 0 {
+            // We're done with migration
+            break;
+        }
+        if num_posts == new_num_posts && num_posts > 0 {
+            // Something went wrong, so we stop and let the state locked.
```

Comment:
Wdyt about moving the failed posts into a new `state.failed_posts` map and then continue migration with the remaining posts? Once the migration is done, we could move `state.failed_posts` to `state.posts`, clear `read_only` flag and attempt a new upgrade via DAO even if some posts have failed (assuming the number of failed posts is low enough)?

Reply:
Yes, this is a great idea, will add!